package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -196,3 +294,94 @@ def plot(self, ax=None, **kwargs):
         a = self.angles.reshape(np.prod(self.angles.shape[:2]), 2)
         ax.scatter(a[:,0], a[:,1], **kwargs)
         return ax
+
+class Janin(Ramachandran):
+    """Calculate chi1 and chi2 dihedral angles of selected residues.
+
+    Chi1 and chi2 angles will be calculated for each residue corresponding to
+    `atomgroup` for each time step in the trajectory. A :class:`~MDAnalysis.ResidueGroup`
+    is generated from `atomgroup` which is compared to the protein to determine
+    if it is a legitimate selection.
+
+    Note
+    ----
+    If the residue selection is beyond the scope of the protein, then an error
+    will be raised. If the residue selection includes the residues ALA, CYS,
+    GLY, PRO, SER, THR, or VAL, then a warning will be raised and they will be
+    removed from the list of residues, but the analysis will still run. Some
+    topologies have altloc attribues which can add duplicate atoms to the
+    selection and must be removed.
+
+    """
+    def __init__(self, atomgroup, **kwargs):
+        """Parameters
+        ----------
+        atomgroup : AtomGroup or ResidueGroup
+            atoms for residues for which chi1 and chi2 are calculated
+
+        Raises
+        ------
+        ValueError
+             If the selection of residues is not contained within the protein
+
+        ValueError
+             If not enough or too many atoms are found for a residue in the
+             selection, usually due to missing atoms or alternative locations
+
+        """
+        super(Ramachandran, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+        residues = atomgroup.residues
+        protein = atomgroup.universe.select_atoms("protein").residues
+        remove = residues.atoms.select_atoms("resname ALA CYS GLY PRO SER"
+                                             " THR VAL").residues
+
+        if not residues.issubset(protein):
+            raise ValueError("Found atoms outside of protein. Only atoms "
+                             "inside of a 'protein' selection can be used to "
+                             "calculate dihedrals.")
+        elif len(remove) != 0:
+            warnings.warn("All ALA, CYS, GLY, PRO, SER, THR, and VAL residues"
+                          " have been removed from the selection.")
+            residues = residues.difference(remove)
+
+        self.ag1 = residues.atoms.select_atoms("name N")
+        self.ag2 = residues.atoms.select_atoms("name CA")
+        self.ag3 = residues.atoms.select_atoms("name CB")
+        self.ag4 = residues.atoms.select_atoms("name CG CG1")
+        self.ag5 = residues.atoms.select_atoms("name CD CD1 OD1 ND1 SD")
+
+        if any(len(self.ag1) != len(ag) for ag in [self.ag2, self.ag3,
+                                                   self.ag4, self.ag5]):
+            raise ValueError("Too many or too few atoms selected. Check for "
```

Comment:
Similar to `Ramachandran`, altloc will make the selections fail, so here the class simply fails. Unlike `Ramachandran`, the user can then select for only on altloc and the analysis will run. Also I've found some topologies that were incomplete and did not have all the atoms it was supposed to so that's why it says 'too few'.

Reply:
altloc (and anything else that specifies as "dual topology") is a problem. 

@richardjgowers is there a sensible way how a user can switch between different topologies? The problem is that at the moment, `ag.residues` will contain _all_ atoms, including altloc ones, even if my selection for `ag` excluded them. This is problematic for the `residue.phi_selection()`, `residue.psi_selection()` and `residue.chi1_selection()` methods, which then see groups that are not 4 atoms.

Reply:
Currently, not really.  This would have to be something done when the file is parsed.  It sounds like maybe `Residue.phi_selection()` should take an altloc kwarg or something to tell it what to identify?

Reply:
MMTF switches between topologies... somehow, so we have that.

We could/should use the fact that the `residue.X_selections()` are methods. Perhaps indeed use a default `Residue.phi_selection(selection="altloc ' '")` or similar – with the issue that we also have to catch the case when this selection makes no sense.

Reply:
I think mmtf does the topologies as frames, (or rather we read models as frames, like how PDB is abused (or at least one of the various ways..)).  

Maybe naïve question, is altloc just a different position? Or can it be different functionalisation?  If the former, I guess you could write an altloc aware reader/parser which expands out the different possibilities.

Reply:
My understanding is that altloc just refers to different positions, but that those positions correspond to separate atoms that are both present in the topology, so they are treated just like extra atoms that are part of the residue. One of the issues with altloc selection is that there doesn't seem to be a way to select the atoms that have no altloc (at least I haven't figured it out). So to get the atoms with no altloc and those with altloc A, you have to say 'not altloc B', assuming there is only A and B for options. I think there can be more altlocs but I have yet to see any that go into C or higher.

Reply:
Do you have the PDB ID of a file with altloc handy?

If it has altlocs then we assign a default, which is either `''` (empty string) or `' '` (one space). I don't know if our selection language can deal with it. But there's always the object-oriented/pandas-style approach
```python
ag[ag.altloc = '']
```
where you use boolean indexing. (Assuming that we expose an `.altloc` attribute somewhere... @richardjgowers would know.)

Reply:
I've been using PDB ID 19hc, but that method returns an AttributeError and says that `AtomGroup` has no altloc attribute

Reply:
@richardjgowers was right
```python
>>> import MDAnalysis as mda
>>> u = mda.fetch_mmtf('19hc')
>>> u.atoms.altLocs
array(['', '', '', ..., '', '', ''], dtype=object)
>>> set(u.atoms.altLocs)
{'', 'A', 'B'}
# select no A or B
>>> u.atoms[u.atoms.altLocs == '']
<AtomGroup with 5944 atoms>
>>> u.atoms
<AtomGroup with 6098 atoms>
```
The default altLoc indicator is the empty string.